src/infrastructure/implementations/blockstack-crux-user-repository.ts

```diff
@@ -99,33 +107,41 @@ export class BlockstackCruxUserRepository implements ICruxUserRepository {
                 enabledParentAssetFallbacks: [],
             },
             privateAddresses: {},
-            publicKey: await keyManager.getPubKey(),
         };
+        let cruxpayPubKey = await keyManager.getPubKey();
         if (cruxUserInformation.registrationStatus.status === SubdomainRegistrationStatus.DONE) {
-            const cruxpayObject: ICruxpayObject = await this.getCruxpayObject(cruxID);
+            const cruxpayJson = await this.getCruxpayObjectAndPubKey(cruxID);
+            const cruxpayObject: ICruxpayObject = cruxpayJson.cruxpayObject;
+            cruxpayPubKey = cruxpayJson.pubKey;
             const dereferencedCruxpayObject = this.dereferenceCruxpayObject(cruxpayObject);
             addressMap = dereferencedCruxpayObject.addressMap;
             if (dereferencedCruxpayObject.cruxUserData) {
                 cruxUserData = dereferencedCruxpayObject.cruxUserData;
             }
         } else if (cruxUserInformation.registrationStatus.status === SubdomainRegistrationStatus.PENDING) {
-            const cruxpayObject: ICruxpayObject = await this.getCruxpayObject(cruxID, undefined, publicKeyToAddress(await keyManager.getPubKey()));
+            const cruxpayJson = await this.getCruxpayObjectAndPubKey(cruxID, undefined, publicKeyToAddress(await keyManager.getPubKey()));
+            const cruxpayObject: ICruxpayObject = cruxpayJson.cruxpayObject;
+            cruxpayPubKey = cruxpayJson.pubKey;
             const dereferencedCruxpayObject = this.dereferenceCruxpayObject(cruxpayObject);
             addressMap = dereferencedCruxpayObject.addressMap;
             if (dereferencedCruxpayObject.cruxUserData) {
                 cruxUserData = dereferencedCruxpayObject.cruxUserData;
             }
         }
-        return new CruxUser(cruxID, addressMap, cruxUserInformation, cruxUserData);
+        return new CruxUser(cruxID, addressMap, cruxUserInformation, cruxUserData, cruxpayPubKey);
     }
     public save = async (cruxUser: CruxUser, keyManager: IKeyManager): Promise<CruxUser> => {
         const cruxpayObject = this.constructCruxpayObject(cruxUser.getAddressMap(), cruxUser.info, cruxUser.config, await keyManager.getPubKey(), cruxUser.privateAddresses);
```

Comment:
remove the publicKey from this path if this is not being used. Also inside `this.constructCruxpayObject`

Reply:
Removed pubKey from __userData__